test/e2e/framework/util.go

```diff
@@ -1246,33 +1257,63 @@ func WaitForPodNoLongerRunningInNamespace(c clientset.Interface, podName, namesp
 }
 
 func WaitTimeoutForPodNoLongerRunningInNamespace(c clientset.Interface, podName, namespace, resourceVersion string, timeout time.Duration) error {
-	w, err := c.Core().Pods(namespace).Watch(metav1.SingleObject(metav1.ObjectMeta{Name: podName, ResourceVersion: resourceVersion}))
-	if err != nil {
-		return err
+	return wait.PollImmediate(10*time.Millisecond, timeout, podCompleted(c, podName, namespace))
```

Comment:
Not sure if bumping timeout to 15 minutes would help? We saw `wait.ErrWaitTimeout` from `watch.Until` within 5 minutes, which indicates the watch closed before timeout.